tests/optimize/test_objective.py

```diff
@@ -74,5 +78,125 @@ def tearDown(self):
+        self.epsilon = relentless.variable.DesignVariable(value=1.0)
+        self.sigma = relentless.variable.DesignVariable(value=0.9)
+        lj.coeff['1','1'].update({'epsilon':self.epsilon, 'sigma':self.sigma, 'rmax':2.7})
+        self.potentials = relentless.simulate.Potentials(pair_potentials=lj)
+        self.potentials.pair.rmax = 3.6
+        self.potentials.pair.num = 1000
+        self.potentials.pair.fmax = 100.
+
+        v_obj = relentless.volume.Cube(L=10.)
+        self.target = relentless.ensemble.Ensemble(T=1.5, V=v_obj, N={'1':50})
+        rs = np.arange(0.5*self.dr, 5.0, self.dr)
+        gs = np.exp(-self.target.beta*lj.energy(('1','1'),rs))
+        self.target.rdf['1','1'] = relentless.ensemble.RDF(r=rs, g=gs)
+
+        self.thermo = relentless.simulate.dilute.AddEnsembleAnalyzer()
+        self.simulation = relentless.simulate.dilute.Dilute(operations=[self.thermo])
+
+    def test_init(self):
+        relent = relentless.optimize.RelativeEntropy(self.target,
+                                                     self.simulation,
+                                                     self.potentials,
+                                                     self.thermo,
+                                                     self.dr)
+        self.assertEqual(relent.target, self.target)
+        self.assertEqual(relent.simulation, self.simulation)
+        self.assertEqual(relent.potentials, self.potentials)
+        self.assertEqual(relent.thermo, self.thermo)
+        self.assertAlmostEqual(relent.dr, self.dr)
+        self.assertIsNone(relent.communicator)
+
+        #test communicator argument
+        comm = relentless.mpi.world
+        relent.communicator = comm
+        self.assertEqual(relent.target, self.target)
+        self.assertEqual(relent.simulation, self.simulation)
+        self.assertEqual(relent.potentials, self.potentials)
+        self.assertEqual(relent.thermo, self.thermo)
+        self.assertAlmostEqual(relent.dr, self.dr)
+        self.assertEqual(relent.communicator, comm)
+
+        #test invalid target ensemble
+        with self.assertRaises(ValueError):
+            relent.target = relentless.ensemble.Ensemble(T=1.5, P=1, N={'1':50})
+
+    def relent_grad(self, var):
+        sim = self.simulation.run(self.target, self.potentials, directory=None)
+        sim_ens = self.thermo.extract_ensemble(sim)
+        rs = self.potentials.pair.r[1:]
+        dus = self.potentials.pair.derivative(('1','1'),var)[1:]
+        dudvar = relentless._math.Interpolator(rs,dus)
+        sim_factor = sim_ens.N['1']**2*sim_ens.beta/(sim_ens.V.volume*self.target.V.volume)
+        tgt_factor = self.target.N['1']**2*self.target.beta/(self.target.V.volume**2)
+
+        r = np.arange(0.05, 3.55, self.dr)
+        y = 2*np.pi*r**2*(sim_factor*sim_ens.rdf['1','1'](r)-tgt_factor*self.target.rdf['1','1'](r))*dudvar(r)
+        return scipy.integrate.trapz(x=r, y=y)
```

Comment:
This is a nice idea, but I think it would be good to make this function a little less dependent on some of the internals of the relentless code. For example, you could hard-code the integrands to the functional forms you expect (rather than use the simulation & ensemble). We want to try to get as independent of a verification as possible.